addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BridgeHandler.java

```diff
@@ -0,0 +1,219 @@
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.wso2iots.internal.config.BridgeConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is responsible for handling commands, which are
+ * sent to the channels.
+ *
+ * @author Ramesha Karunasena - Initial contribution
+ */
+public class BridgeHandler extends BaseBridgeHandler {
+
+    private ScheduledFuture<?> refreshJob;
+
+    private final Logger logger = LoggerFactory.getLogger(BridgeHandler.class);
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Initializing bridge handler.");
+
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        logger.debug("config hostname = {}", config.hostname);
+        logger.debug("config port = {}", config.port);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.hostname) == null) {
+            errorMsg = "Parameter 'hostname' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.port == null) {
+            errorMsg = "Parameter 'port' must be configured";
+            validConfig = false;
+        }
+        if (config.refresh == null) {
+            errorMsg = "Parameter 'refresh' must be configured";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            updateStatus(ThingStatus.OFFLINE);
+            startAutomaticRefresh(config);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the bridge handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No channels - nothing to do
+    }
+
+    /**
+     * Start the job refreshing the handler
+     */
+    private void startAutomaticRefresh(BridgeConfiguration config) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+
+                    checkConnection();
+
+                } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception